test/python/custom_op_shape_dtype_fn.py

```diff
@@ -0,0 +1,71 @@
+import os
+import tempfile
+
+import torch
+import torch.utils.cpp_extension
+import torch_mlir
+from torch_mlir_e2e_test.annotations import export, annotate_args
+
+
+# RUN: %PYTHON %s | FileCheck %s
+
+
+def identity(_5: torch.Tensor):
+    return _5
+
+
+goofy_lib = torch.library.Library("goofy", "DEF")
+goofy_lib.define("identity(Tensor t) -> Tensor")
+goofy_lib.impl("identity", identity)
+
+
+class CustomOpExampleModule(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args(
+        [
+            None,
+            ([-1, -1], torch.float32, True),
+        ]
+    )
+    def forward(self, a):
+        b = 2 * a
+        return torch.ops.goofy.identity(b)
+
+
+mod = CustomOpExampleModule()
+mod.eval()
+
+abstract_interp_src = """\
+func.func @__torch_mlir_shape_fn.goofy.identity(%arg0: !torch.list<int>) -> !torch.list<int> {
+  return %arg0 : !torch.list<int>
+}
+func.func @__torch_mlir_dtype_fn.goofy.identity(%arg0: !torch.int, %arg1: !torch.int) -> !torch.int {
+  return %arg1 : !torch.int
+}
+func.func @__torch_mlir_has_value_semantics.goofy.identity() { return }
+"""
+
+with tempfile.NamedTemporaryFile(mode="w") as tmp:
+    tmp.write(abstract_interp_src)
+    tmp.flush()
+    module = torch_mlir.compile(
+        mod,
+        torch.ones(3, 4),
+        output_type="torch",
+        backend_legal_ops=["torch.operator"],
```

Comment:
We need to change the handling of backend_legal_ops to treat torch.operator specially, since otherwise this just opens up a weird can of worms suddenly allowing all torch.operator's through.

I think we should cahnge it so that the names passed in backend_legal_ops should always be in the sense of the PyTorch operator registry namespace (currently they are prefixed with `torch.`). I think the best path forward here is to change the handling of backend_legal_ops to not need the `torch.` prefix.

That is, `backend_legal_ops=["aten.foo", "goofy.identity"]` would be how we use it and it wouldn't matter if it was an op we know about or a torch.operator. If it is torch.operator then we check the name attribute, and if it is a different `torch` dialect op then we check that the op name is `torch.goofy.identity` etc.

That will require stripping the torch. prefix from a couple places like here: https://github.com/llvm/torch-mlir/blob/f2a05f2dc0dade5cf03f23b317835b36391e0c2d/python/torch_mlir/__init__.py#L243 but I think this puts us in a much better place.